### ISSUE TITLE
Psion Revo (German) rom dump

### ISSUE THREAD

**Issue by CO-Windler** [explit7/Psion-ROM#2]
This is the firmware of my Psion Revo (German version).

PsiROMx shows dump defaults:

Start address 
50000000
Size (KB) 8192
End address 
50800000

But it has not enough RAM to dump it in one piece. I found out that the address range can be set in the menu, so I dumped 2 4MB chunks, those I combined with HxD, which looks correct.

sys$rom1.bin = 50000000 to 50400000
sys$rom2.bin = 50400000 to 50800000

[Psion Revo (German)_firmware.zip](https://github.com/explit7/Psion-ROM/files/11231773/Psion.Revo.German._firmware.zip)

**Comment by explit28** [comment]
great!
could you create an merge request to upload the Revo ROM?

**Comment by CO-Windler** [comment]
How can I do this? Of course I want to make it available. This is a multilingual firmware version that contains many European languages. The keyboard layout is German (QWERTZ).

This is how and why I had dumped it:
https://github.com/explit7/Psion-ROM/issues/1

I have merged the 2 halves (each 4MB) into an 8MB file, but for re-installing on a Revo with 8MB RAM it will need to be split into halves again.

**Comment by explit28** [comment]
Hi, just make a commit on Github with the one big file and post how to split them in 2 halves.

Just google how to commit files on github, or ask chatgpt.

Its not complex, but i had also some problems first time.

Thank you for your work!
Cheers!

ExPLIT

On Saturday, 15 April 2023, CYBERYOGI =CO=Windler wrote:
> How can I do this? Of course I want to make it available. This is a multilingual firmware version that contains many European languages. The keyboard layout is German (QWERTZ).
> 
> I have merged the 2 halves (each 4MB) into an 8MB file, but for re-installing on a Revo with 8MB RAM it will need to be split into halves again.
> 
> -- 
> Reply to this email directly or view it on GitHub:
> https://github.com/explit7/Psion-ROM/issues/2#issuecomment-1509470574
> You are receiving this because you commented.
> 
> Message ID: ***@***.***

-- 

Kind regards / Mit freundlichen Grüßen

ExPLIT IT Soltions
Pawel Radomychelski

**Comment by CO-Windler** [comment]
> Hi, just make a commit on Github with the one big file and post how to split them in 2 halves. Just google how to commit files on github, or ask chatgpt. Its not complex, but i had also some problems first time. Thank you for your work! Cheers! ExPLIT

When I select in Firefox "Add file", I only get a popup message "File uploads are disabled."

I already had attached the rom file. Please add it yourself. To split it into halves, you can use e.g. the HxD hex editor. 

CAUTION: I haven't tried if it can be reinstalled this way on an 8MB Revo, or if overwriting only one half of the rom will brick it without chance of replacing the 2nd halve.

According to wxHexEditor the actual rom contents seems to end at 740C3Fh, so I now have removed here the "FF" section at the end of the file to shorten it to about 7.3MB. I don't know if this will be sufficient to reinstall it. If not, it may be still necessary to split it in 2 parts to install them separately with PsiROM at their original addresses.. When making the first halve 4MB, this would mean

sys$rom1.bin = 50000000 to 50400000
sys$rom2.bin = 50400000 to 50800000

[Psion Revo (German)_firmware shorter.zip](https://github.com/explit7/Psion-ROM/files/11241206/Psion.Revo.German._firmware.shorter.zip)

**Comment by explit28** [comment]
No, not in Firefox.
Git is working with commands from commandline.

If you are not familiar with git - just send me the files and readme.txt and i will make an git commit.

Cheers

ExPLIT



On Sunday, 16 April 2023, CYBERYOGI =CO=Windler wrote:
> When I select in Firefox "Add file", I only get a popup message "File uploads are disabled."
> 
> -- 
> Reply to this email directly or view it on GitHub:
> https://github.com/explit7/Psion-ROM/issues/2#issuecomment-1510032973
> You are receiving this because you commented.
> 
> Message ID: ***@***.***

-- 

Kind regards / Mit freundlichen Grüßen

ExPLIT IT Soltions
Pawel Radomychelski

**Comment by CO-Windler** [comment]
> No, not in Firefox. Git is working with commands from commandline. If you are not familiar with git - just send me the files and readme.txt and i will make an git commit. Cheers ExPLIT

Huh? The zip files are already here in the attachment of these postings.

I am an archivist, not a Psion OS expert. I backupped the firmware for preservation (emulation or future repairs if original chips will suffer of bitrot) and for research of the inner working, so I don't know if this rom dump can be safely reinstalled or will brick the device by too small RAM - and I am not really keen to find it out. (I am happy enough that I managed to replace the leaked battery pack with an external battery box (safer for longterm storage as a collectable), and I had to reflow the solder to get the LCD screen back to work, so I could finally install PsiWin and PsiROMx to extract the files. Currently there is no battery inserted and I don't want to spend more time with risky flashloader experiments.)

Yet I use Git only for bugreporting and once I installed a PulseView nightly built on Linux (typing commands from a website), but I am not familiar with it.
